### PR TITLE
feat(delegation-api): add custom delegation scope field to delegation index

### PR DESCRIPTION
**Pull request by Herdismaria**
[Delegation index pitch](https://www.notion.so/Delegation-Index-5aa41671d64d4031b276d45423b11741?pvs=4)

## What
Add scopes field to delegation index to keep track of api scopes for custom delegations.

## Why
To facilitate delegation look up.

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

### CODE REVIEW

File: apps/services/auth/ids-api/src/app/delegations/test/delegation-index.spec.ts [hunk 1]
```diff
@@ -306,6 +306,98 @@ describe('DelegationsIndexService', () => {
...
+        const fromNationalId = testcase.customDelegations[0].fromNationalId
+        const testDelegation = await delegationModel.findOne({
+          where: {
+            fromNationalId,
+            toNationalId: user.nationalId,
+          },
+        })
+
+        // Add remove scope from delegation
+        if (testDelegation) {
```
Comment (baering) [inline apps/services/auth/ids-api/src/app/delegations/test/delegation-index.spec.ts:323]: Wondering why there is an if statement here and in line 366. Do you expect this to not exist at some point?

File: libs/auth-api-lib/src/lib/delegations/delegations-index.service.ts [hunk 2]
```diff
@@ -24,6 +29,18 @@ type SortedDelegations = {
...
+const hasAllSameScopes = (a: string[] | undefined, b: string[] | undefined) => {
+  // Only custom delegations have scopes and they are never undefined
+  if (!a && !b) {
+    return true
+  }
+
+  if (a?.length !== b?.length) {
+    return false
+  }
+  return a?.every((s) => b?.includes(s))
```
Comment (baering) [inline libs/auth-api-lib/src/lib/delegations/delegations-index.service.ts:41]: `=== true` so it doesn't return undefined in any case?

**Review by baering — APPROVED** [review]
LGTM ✅

File: apps/services/auth/ids-api/src/app/delegations/test/delegation-index.spec.ts (hunk 1 above)
Comment (Herdismaria) [reply apps/services/auth/ids-api/src/app/delegations/test/delegation-index.spec.ts:323]: Just to make typescript happy, `delegationModel.findOne` can return null 🤷‍♀️

File: libs/auth-api-lib/src/lib/delegations/delegations-index.service.ts (hunk 2 above)
Comment (Herdismaria) [reply libs/auth-api-lib/src/lib/delegations/delegations-index.service.ts:41]: Will fix 👍

**Pull request merged**